src/Microsoft.Azure.WebJobs.Host/Bindings/BindingContext.cs

```diff
@@ -110,5 +133,14 @@ public ValueBindingContext ValueContext
                 return _valueContext;
             }
         }
+
+        public async Task<IReadOnlyDictionary<string, object>> GetBindingDataAsync()
```

Comment:
One of the big perf improvements I made for Http was to make resolution of binding data lazy, so it doesn't happen if not needed. E.g. in most http scenarios, binding data isn't needed since no binding expressions or in/out bindings are used. So many of the changes in this PR are about plumbing async lazy resolution of binding data through the SDK.